src/main/java/seedu/recipe/model/recipe/RecipePortion.java

```diff
@@ -1,95 +1,133 @@
+     * @param portionUnit The portion unit instance
+     */
+    public RecipePortion(int lowerRange, int upperRange, PortionUnit portionUnit) {
+        if (upperRange < 0) {
+            checkArgument(
+                    isValidRecipePortion(String.format("%s %s", lowerRange, portionUnit.toString())),
+                    MESSAGE_CONSTRAINTS);
+        } else {
+            checkArgument(
+                    isValidRecipePortion(
+                            String.format("%s-%s %s", lowerRange, upperRange, portionUnit.toString())),
+                    MESSAGE_CONSTRAINTS);
+        }
         this.lowerRange = lowerRange;
         this.upperRange = upperRange;
         this.portionUnit = portionUnit;
     }
 
     @Override
     public String toString() {
-        StringBuilder sb = new StringBuilder();
-        sb.append(lowerRange).append("-").append(upperRange).append(" ").append(portionUnit.toString());
-        return sb.toString();
+        if (upperRange > 0) {
+            return String.format(
+                    "%s - %s %s",
+                    lowerRange, upperRange, portionUnit.toString()
+            );
+        }
+        return String.format("%s %s", lowerRange, portionUnit.toString());
     }
 
     @Override
     public int hashCode() {
         return Objects.hash(lowerRange, upperRange, portionUnit);
     }
 
+    public int getLowerRange() {
+        return lowerRange;
+    }
+
+    public int getUpperRange() {
+        return upperRange;
+    }
+
+    public PortionUnit getPortionUnit() {
+        return portionUnit;
+    }
+
+    /**
+     * Checks if the provided String is formatted properly and can be split into the appropriate tokens to
+     * generate a RecipePortion instance.
+     * @param test The string to test.
+     * @return The boolean indicating if it is valid.
+     */
     public static boolean isValidRecipePortion(String test) {
-        if (test.matches(VALIDATION_REGEX)) {
-            String[] tokens = test.split(TOKENIZE_REGEX);
-            if (tokens.length != 3) {
-                return false;
-            }
-            try {
-                int lower = Integer.parseInt(tokens[0]);
-                int upper = Integer.parseInt(tokens[1]);
-                if (upper < lower) {
-                    return false;
-                }
-            } catch (NumberFormatException e) {
-                return false;
-            }
-            return true;
+        if (!test.matches(VALIDATION_REGEX)) {
+            return false;
         }
-        return false;
+        String[] tokens = test.split(TOKENIZE_REGEX);
+        if (tokens.length != 3) {
+            String[] singularToken = test.split("\\s+");
+            int lower = Integer.parseInt(singularToken[0]);
+            return lower > 0;
+        }
+        int lower = Integer.parseInt(tokens[0]);
+        int upper = Integer.parseInt(tokens[1]);
+        return upper > lower;
```

Comment:
code quality: this should probably in the `else` case too since it is an alternative path depending on the value of `tokens.length`